### PR TITLE
GH-484, GH-482 - image with content component fixes, and styling updates

### PR DESCRIPTION
**Pull request by jhung**
* [x] This pull request has been tested by running `npm run test` without errors
* [x] This pull request has been built by running `npm run build` without errors
* [x] This isn't a duplicate of an existing pull request

Fixes:
- #484 : image width scale now applies to individual images
- #482 : added default Center alignment to new images
- Added a fix for an unreported issue where if Scale Image Width is 100% and there is content, causing the content to wrap onto the next block. If Scale Image Width is 100% and content is not empty, Scale Image Width will now be set to 75%.
- image position with text help text improvements
- removed max image height. It wasn't being used much and was confusing when used with Scale Image Width.

## Steps to test #482

1. Go to a Project and add a image with text component to the content
2. Set Scale Image Width to 25%
3. Publish
4. Notice that the output will align image to the center at 25%

## Steps to test #484 

1. Create a new Project
2. Switch to Markdown view for the Body
3. Copy and paste one of the [test cases located here](https://drive.google.com/drive/folders/1kjTi-cIGoBW_pk175ZqqecZAVLgf1yBV?usp=drive_link).
4. The image should be aligned and scaled as the user specified, with the exception of 2 edge cases:
   1. If the width scaling is 100% and there is accompanying content, then the width is reduced to 75%.
   2. If the image center aligned and there is accompanying content, then it will change to left aligned. 

## Steps to test Scale Image Width 100% with Content

1. Go to a Project and add a image with text component to the content
6. Set Scale Image Width to 100%
7. Set Content to some string value (don't leave it empty)
8. Publish
9. View the page in a browser - notice how the image is 75% width and the content is aligned in the remaining 25%.

### CODE REVIEW

File: src/shortcodes/image-position-with-text.js [hunk 1]
```diff
@@ -2,15 +2,14 @@ const markdownFilter = require('../filters/markdown.js');
...
 
-	let output;
-	output = `<style>:root {--image-position-max-height: ${maxHeight}; --image-position-width: ${scale};}</style>`;
 	if (content === '') {
-		output += `<img src="${image}" alt="${alt}" class="image-position--${imagePosition} image-position--image-height" />`;
+		output += `<p><img src="${image}" alt="${alt}" class="image-position--${imagePosition} image-position--image-height  image-and-text__scale--${scale} image-and-text__max-height--${maxHeight}" /></p>`;
 	} else {
-		imagePosition = imagePosition.trim() === 'center' ? 'left' : imagePosition;
-		output += `<div class="image-and-text image-and-text--image-${imagePosition} image-and-text--vertical-${verticalAlignment}"><figure class="image-and-text__media"><img src="${image}" alt="${alt}" class="image-position--image-height" /></figure><div class="image-and-text__content">${markdownFilter(content)}</div></div>`;
+		scale = scale === '100' ? '75' : scale; // Reduce to 75% width to allow content to be shown beside it.
```
[the remark this answers is not in the document]
Comment (greatislander) [reply src/shortcodes/image-position-with-text.js:10]: It does appear that there are different rules at larger breakpoints, see: https://github.com/jhung/idrc/blob/92f36cc6045e625c58cf4ba923767e4cd828bdb2/src/assets/styles/vendors/_cms-editor-components.scss#L136-144

[the remark this answers is not in the document]
Comment (chosww) [reply src/shortcodes/image-position-with-text.js:10]: oh, I meant classes for displaying text to left and right of the image (there is no .image-and-text--image-left--100 or .image-and-text--image-right--100). This is because to display text next to the image, text need some spaces next to it. I think we shouldn't have 100 option at all for these cases.

[the remark this answers is not in the document]
Comment (jhung) [reply src/shortcodes/image-position-with-text.js:10]: @chosww and @greatislander I was rethinking this situation where the user has specified 100% width for an image and has also specified some text / content. In this situation it might be less confusing to have the image 100% width in a row above the content.

Thoughts?

[the remark this answers is not in the document]
Comment (chosww) [reply src/shortcodes/image-position-with-text.js:10]: I think that makes sense @jhung, and that would be clearly distinguishable with 75% option.

File: src/utils/generateProjectPermalink.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ module.exports = (data) => {
 	}
 
 	const locale = data.locale;
-	const localeSlug = config.languages[locale].slug || locale;
+	const localeSlug = config.languages[locale]?.slug || locale;
```
Comment (chosww) [inline src/utils/generateProjectPermalink.js:13]: @greatislander the issue Jon was having was caused by this line, because config.languages[locale] was undefined. This was the fix I suggested, but if you have different opinion, please let me know.

Comment (greatislander) [reply src/utils/generateProjectPermalink.js:13]: That looks like a great approach, thanks @chosww and @jhung!

**Review by chosww — APPROVED** [review]
